linetools/spectra/lsf.py

```diff
@@ -0,0 +1,295 @@
+"""
+Module for dealing with LSFs of various astronomical instruments.
+"""
+from __future__ import print_function
+
+import numpy as np
+from scipy.interpolate import interp1d
+from astropy.io import fits, ascii
+from astropy.units import Quantity
+from astropy.table import Table, QTable, Column
+# import matplotlib.pyplot as plt
+import astropy.units as u
+import glob, imp
+
+lt_path = imp.find_module('linetools')[1]
+
+class LSF(object):
+
+    """Class to deal with line-spread-functions (LSFs) from
+    various different astronomical spectrographs.
+
+    Note: only implemented for HST/COS at the moment.
+
+    Parameters
+    ----------
+    instr_config : dict
+        A dictionary with the instrument configuration details relevant
+        to the required LSF. Mandatory keywords of the dict are: ['name'], 
+        all of which must be either string or None. 
+        Note: There must be extra relevant keywords specific to each instrument.
+
+    """
+
+    def __init__(self, instr_config):
+        #mandatory keys for characterizing a spectrograph mode
+        self.mandatory_dict_keys = ['name']
+                
+        #Check correct format
+        if not isinstance(instr_config,dict):
+            raise SyntaxError('`instr_config` must be a dictionary.')
```

Comment:
I don't think `SyntaxError` is the right thing to raise here. `TypeError` would be more appropriate (https://docs.python.org/2/library/exceptions.html#exceptions.TypeError). Or just `RuntimeError` if you don't want to have to worry about it.